packages/time-utilities/src/lib/FriendlyDuration.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * The supported time types
+ */
+export const enum TimeTypes {
+	Second = 'SECOND',
+	Minute = 'MINUTE',
+	Hour = 'HOUR',
+	Day = 'DAY',
+	Week = 'WEEK',
+	Month = 'MONTH',
+	Year = 'YEAR'
+}
+
+/**
+ * The duration of each time type in milliseconds
+ */
+const kTimeDurations: readonly [TimeTypes, number][] = [
+	[TimeTypes.Year, 31536000000],
+	// 29.53059 days is the official duration of a month: https://en.wikipedia.org/wiki/Month
+	[TimeTypes.Month, 2628000000],
+	[TimeTypes.Week, 1000 * 60 * 60 * 24 * 7],
+	[TimeTypes.Day, 1000 * 60 * 60 * 24],
+	[TimeTypes.Hour, 1000 * 60 * 60],
+	[TimeTypes.Minute, 1000 * 60],
+	[TimeTypes.Second, 1000]
+];
+
+/**
+ * Display the duration
+ * @param duration The duration in milliseconds to parse and display
+ * @param assets The language assets
+ */
+export const friendlyDuration = (duration: number, assets: DurationFormatAssetsTime, precision = 7) => {
```

Comment:
Make this a class that accepts `DurationFormatAssetsTime` in the constructor.

Also rename this to something else, e.g. `DurationFormatter`, and `assets` to something such as `units`.